### PR TITLE
added support for app/system info sections

### PR DESCRIPTION
**Pull request by beniwohli**
this also adds configuration options for git_ref and app version, and removes deprecated handling of `async` keyword argument (replaced by `async_mode`)

### CODE REVIEW

File: opbeat/base.py [hunk 1]
```diff
@@ -337,10 +337,10 @@ def build_msg_for_logging(self, event_type, data=None, date=None,
...
 
     def build_msg(self, data=None, **kwargs):
-        data.setdefault('machine', {'hostname': self.hostname})
-        data.setdefault('organization_id', self.organization_id)
-        data.setdefault('app_id', self.app_id)
-        data.setdefault('secret_token', self.secret_token)
+        data = data or {}
+        data['app'] = self.get_app_info()
+        data['system'] = self.get_system_info()
+        data.update(**kwargs)
```
Comment (ruflin) [inline opbeat/base.py]: Feels a bit too powerful :-)

File: opbeat/base.py [hunk 2]
```diff
@@ -643,23 +639,49 @@ def _traces_collect(self):
...
-            platform_bits['framework'] = self._framework_version
-        return ' '.join('%s=%s' % item for item in platform_bits.items())
-
+    def get_app_info(self):
+        language_version = platform.python_version()
+        if hasattr(sys, 'pypy_version_info'):
+            runtime_version = '.'.join(map(str, sys.pypy_version_info[:3]))
+        else:
+            runtime_version = language_version
+        return {
```
Comment (ruflin) [inline opbeat/base.py]: Nice. This makes it very easy to see what is constructed.

File: tests/contrib/django/django_tests.py [hunk 3]
```diff
@@ -643,7 +643,7 @@ def test_transaction_metrics(self):
             assert len(transactions) == 1
             transaction = transactions[0]
             assert transaction['duration'] > 0
-            assert transaction['result'] == 200
+            assert transaction['result'] == '200'
```
Comment (ruflin) [inline tests/contrib/django/django_tests.py]: Could be also a int? Or will there be non int data here?

Comment (beniwohli) [inline tests/contrib/django/django_tests.py]: Yes, non-http transactions will have non-http-status results.

**Pull request merged**